src/traffic_server/InkAPI.cc

```diff
@@ -10321,3 +10339,86 @@ TSDbgCtlCreate(char const *tag)
 
   return DbgCtl::_get_ptr(tag);
 }
+
+namespace rpc
+{
+extern std::mutex g_rpcHandlingMutex;
+extern std::condition_variable g_rpcHandlingCompletion;
+extern ts::Rv<YAML::Node> g_rpcHandlerResponseData;
+extern bool g_rpcHandlerProcessingCompleted;
+} // namespace rpc
+
+tsapi TSRPCProviderHandle
+TSRPCRegister(const char *provider_name, const char *yaml_version)
+{
+  sdk_assert(sdk_sanity_check_null_ptr(yaml_version) == TS_SUCCESS);
+  sdk_assert(sdk_sanity_check_null_ptr(provider_name) == TS_SUCCESS);
+
+  // We want to make sure that plugins are using the same yaml library version as we use internally. Plugins have to cast the TSYaml
+  // to the YAML::Node, in order for them to make sure the version compatibility they need to register here and make sure the
+  // version is the same.
+  if (std::string_view{yaml_version} != YAMLCPP_LIB_VERSION) {
+    return nullptr;
+  }
+
+  rpc::RPCRegistryInfo *info = new rpc::RPCRegistryInfo();
+  info->provider             = provider_name;
+
+  return (TSRPCProviderHandle)info;
+}
+
+tsapi TSReturnCode
+TSRPCRegisterMethodHandler(const char *name, TSRPCMethodCb callback, TSRPCProviderHandle info, const TSRPCHandlerOptions *opt)
+{
+  sdk_assert(sdk_sanity_check_rpc_handler_options(opt) == TS_SUCCESS);
+
+  if (!rpc::add_method_handler_from_plugin(
+        name,
+        [callback](std::string_view const &id, const YAML::Node &params) -> void {
+          std::string msgId{id.data(), id.size()};
```

Comment:
For instance, here you need to create a `std::string` and potentially allocate memory because the view can't be easily passed through. This is why we want to have a consistent view based API.